### PR TITLE
Git cheatsheet

### PR DESCRIPTION
**Pull request by alcantarar**
addresses #4

### CODE REVIEW

File: git_cheatsheet.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Git Cheatsheet
+### Modified from the [Github Education Material](https://education.github.com/git-cheat-sheet-education.pdf)
+### There are additional functions, but this is a basic introduction. 
+#### Note: Text in brackets must be replaced by object described within brackets: "[FIRSTNAME LASTNAME]" -> "Kevin Bacon" 
+## DEFINITIONS
+* **Local Repository** - A version of your repository that physically exists on your computer.
+* **Remote Repository** - A version of your repository that exists on a server (like GitHub's).
+* **Directory** - A folder on your computer. The working directory is the folder you are working in and can be changed using the `cd` command.
+* **Commit** - A snapshot of changes made to files. Files must be staged via the `git add` command. 
+* **Staged/Staging** - The process of identifying which files will be included in a commit. Can include all files in directory or specific files.
```
Comment (GBruening) [inline git_cheatsheet.md]: Not sure we need to talk about this?

File: git_cheatsheet.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+## DIRECTORY NAVIGATION
+Change directory to explicit location:
+```
+cd PATH/TO/DESIRED/DIRECTORY/
```
Comment (GBruening) [inline git_cheatsheet.md:9]: This changes depending on the OS, i think the slashes go the other way for mac/linux

File: git_cheatsheet.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
+# Git Cheatsheet
```
Comment (GBruening) [inline git_cheatsheet.md:1]: Should we tell them that these commands are mostly used in the GitBash app?

File: git_cheatsheet.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+## DIRECTORY NAVIGATION
+Change directory to explicit location:
+```
+cd PATH/TO/DESIRED/DIRECTORY/
+```
+Change directory to a folder "up" one level:
+```
+cd ..
+```
+*This is equivalent to `cd PATH/TO/DESIRED/`*
```
Comment (GBruening) [inline git_cheatsheet.md]: A little unclear, as i'm not sure this specifies a base folder?

cd .. is this only if we actually got into the directory folder

File: git_cheatsheet.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+## CONFIGURATION
+#### Configuring user information used across all local repositories on your computer
+Set a name that is identifiable for credit when reviewing version history:
+```
+git config --global user.name "[FIRSTNAME LASTNAME]"
+```
+Set an email address that will be associated with each commit and required to access private
+private repositories:
```
Comment (GBruening) [inline git_cheatsheet.md]: Private is said twice

File: git_cheatsheet.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+```
+git config --global user.email "[GITHUB EMAIL]"
+```
+## GETTING HELP
+Replace [VERB] with the operation you need help with. `--help` will return the full help page, while `--h` returns a consolidated help page:
+``` 
+git [VERB] --help
+git [VERB] -h
+```
```
Comment (GBruening) [inline git_cheatsheet.md]: We never list what verbs can actually be used here or define verbs

File: git_cheatsheet.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+git diff
+```
+View line-by-line changes made to files that are staged:
+```
+git diff --staged
+``` 
+Finialize the snapshot of your staged file(s) by making a commit:
+```
+git commit -m "[SHORT MESSAGE DESCRIBING CHANGES MADE]"
+```
```
Comment (GBruening) [inline git_cheatsheet.md]: We may want to reorganize these to have the ones we'll use at the top like this one and git push

File: git_cheatsheet.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+git branch
+```
+Create a new branch based on the most recent commit:
+```
+git branch [BRANCH-NAME]
+```
+Switch your local directory to another branch:
+```
+git checkout -b [BRANCH-NAME]
+```
```
Comment (GBruening) [inline git_cheatsheet.md]: the -b i believe creates a new branch, if you remove the -b it changes the branch

File: git_cheatsheet.md (hunk 3 above)
Comment (alcantarar) [reply git_cheatsheet.md:1]: @GBruening they work with powershell/terminal as well.
Edit: but really they are just git functions.

**Review by alcantarar — COMMENTED** [review]
made suggested changes and reorged git functions to be in order of use:
pull, then status, add, commit, and push.

**Comment by alcantarar** [comment]
Slash directions aren't operating system specific in Git. I just checked in the git manual: https://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository

**Comment by GBruening** [comment]
Cool, just wanted to check on the slash direction. I know it can sometimes change between OS's.

**Pull request merged**